src/modules/dashboard/sagas/colonyCreate.ts

```diff
@@ -260,27 +261,61 @@ function* colonyCreate({
       tokenAddress = createAddress(givenTokenAddress);
     }
 
-    /**
-     * If we're not recovering this will be overwritten
+    /*
+     * This is a bit of a cumbersome solution, but should serve fine currently,
+     * until we find a way to properly stabilize IPFS uploads
      */
     let colonyAddress;
     if (createColony) {
-      const colonyMetadataIpfsHash = yield call(
-        ipfsUpload,
-        JSON.stringify({
-          colonyName,
-          colonyDisplayName: displayName,
-          colonyAvatarHash: null,
-          colonyTokens: [],
-        }),
-      );
+      /*
+       * First IPFS upload try
+       */
+      let colonyMetadataIpfsHash;
+      try {
+        colonyMetadataIpfsHash = yield call(
+          ipfsUpload,
+          JSON.stringify({
+            colonyName,
+            colonyDisplayName: displayName,
+            colonyAvatarHash: null,
+            colonyTokens: [],
+          }),
+        );
+      } catch (error) {
+        log.verbose('Could not upload the colony metadata IPFS. Retrying...');
+        log.verbose(error);
+        /*
+         * If the first try fails, then attempt to upload again
+         * We assume the first error was due to a connection issue
+         */
+        colonyMetadataIpfsHash = yield call(
+          ipfsUpload,
+          JSON.stringify({
+            colonyName,
+            colonyDisplayName: displayName,
+            colonyAvatarHash: null,
+            colonyTokens: [],
+          }),
+        );
+      }
```

Comment:
If the second upload call will throw an error, where we're catching this error?

Reply:
the try / catch inside the saga is not about catching the errors, we're just using it as a mechanism for retry-ing.

Both IPFS libs that we use handle their own error catching, we just use the try/catch block for setting the hash value _(what's why we check if it's a string, because if it throws an Error we don't actually know)_

Anyway, this is just a stop-gap solution, until we implement a more proper way of hadling IPFS uploads and their failure states _(most likely via a queue and using Service Workers)_